sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,500 @@
+"response_capacity" properties that can be provided when declaring a
+`task_sequence`.
+
+[source,c++,linenums]
+----
+namespace sycl {
+namespace ext {
+namespace intel {
+namespace experimental {
+
+struct balanced_key {
+  using value_t = property_value<balanced_key>;
+};
+
+struct invocation_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<invocation_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+struct response_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<response_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+inline constexpr balanced_key::value_t balanced;
+
+template <> struct is_property_key<balanced_key> : std::true_type {};
+
+// TODO: Not sure if the template here should just be
+// template<auto &f, class propertiesT>
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<balanced_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+template <uint32_t Size>
+inline constexpr invocation_capacity_key::value_t<Size> invocation_capacity;
+
+template <> struct is_property_key<invocation_capacity_key> : std::true_type {};
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<invocation_capacity_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+template <uint32_t Size>
+inline constexpr response_capacity_key::value_t<Size> response_capacity;
+
+template <> struct is_property_key<response_capacity_key> : std::true_type {};
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<response_capacity_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+} // namespace experimental
+} // namespace intel
+} // namespace ext
+} // namespace sycl
+----
+
+--
+[options="header"]
+|===
+| Property | Description
+| balanced | The `balanced` property is a guarantee to the SYCL device compiler
+that the `task_sequence` object will call exactly the same number of `async` s
+and `get` s over the object's lifetime (i.e. before the `task_sequence`
+destructor is invoked). 
+
+| invocation_capacity | The `invocation_capacity` property defines the minimum
+required number `Size` of `async` invocations that must be invokable without any
+`get` being called on that same set.  This number of `async` invocations without
+a `get` call is the minimum number that will be supported before the `async`
+member function blocks.
+
+| response_capacity | The `response_capacity` property defines the maximum
+number `Size` of outstanding `async` invocations such that all outstanding
+invocations are guaranteed to make forward progress. Further `async` invocations
+beyond `Size` may block until enough `get` calls are invoked such that the
+number of outstanding `async` invocations is reduced to `Size`.
```

Comment:
It's misleading and distracting to refer to calls to 'async' here.  This control isn't about how many outstanding calls you can have before async blocks, that's what invocation_capacity is for.  This control is about how many outstanding calls you can have before you can't be sure that _any_ of them will make forward progress.  I think this example will clarify:

```
use some_pipe::pipe<bool>; 

void foo() {
  some_pipe::write(true);
}

task_sequence<foo, decltype(properties{invocation_capacity<N>, response_capacity<M>})> f;
// N (invocation_capacity) determines the maximum value of num_async
// i.e. num_async <= N to ensure that the call to f::async() doesn't block
for (int i = 0; i < num_async; ++i) {
  f::async();
}
for (int i = 0; i < num_async; i+=block_size) {
  // M (response_capacity) determines the maximum value of block_size
  // i.e. block_size <= M to ensure that the call to some_pipe::read() doesn't block
  for (int j = 0; j < block_size; ++j) {
    some_pipe::read();
  }
  for (int j = 0; j < block_size; ++j) {
    f::get();
  }
}
```

Reply:
The updated wording does not mention asyncs.